help/Readme.md

```diff
@@ -0,0 +1,114 @@
+### General
+The virtual-environment project uses packer tool and contains templates for the following OSs: Windows 2016/2019, Ubuntu 16.04/18.04/20.04. 
+Each image template consists of a set of scripts that are executed on the virtual machine. 
+The Packer process initializes a connection to Azure subscription via Azure CLI, and automatically creates the temporary Azure resources required to build the source VM(temporary resource group, network interfaces, and VM from the "clean" image specified in the template). 
+If deployment succeeded, the build agent connects to the deployed VM and starts to execute installation steps from the selected template.
+In case if any template step fails, image generation will be aborted and temporary VM will be terminated.
+After successfull image generation, snapshot of temporary VM will be converted to VHD image and then uploaded to the specified Azure Storage Account.
+
+### Build Agent requirements
+- `OS` - Windows/Linux
+- `packer` - Can be downloaded from https://www.packer.io/downloads
+- `PowerShell 5.0 or higher` or `PSCore` for linux distributes.
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to a temporary VM packer use WinRM or SSH connections on public IP interfaces.
+If you use a build agent located in an Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case of firewall restrictions, prohibiting connections from public addresses, private virtual network resources can be deployed and passed as arguments to the packer. This approach allows virtual machines to use private connections inside VLAN.
+
+#### Service principal
+Packer uses Service Principal to authorize in Azure infrastructure. To setup image-generation CI or use packer manually — SP with full read-write permissions for selected Azure subscription needed.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+Local machine or [Azure VM](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli) can be used as a build agent.
+
+Download `packer` from https://www.packer.io/downloads, or install it via Chocolately.
+```
+Choco-Install -PackageName packer
```

Comment:
`Choco-Install` doesn't exist on customer's machines. It is our custom helper.
`choco install packer`